src/um/wia.rs

```diff
@@ -0,0 +1,756 @@
+RIDL!{#[uuid(0xa558a866, 0xa5b0, 0x11d2, 0xa0, 0x8f, 0x00, 0xc0, 0x4f, 0x72, 0xdc, 0x3c)]
+interface IWiaDataCallback(IWiaDataCallbackVtbl): IUnknown(IUnknownVtbl) {
+    fn BandedDataCallback(
+        lMessage: LONG,
+        lStatus: LONG,
+        lPercentComplete: LONG,
+        lOffset: LONG,
+        lLength: LONG,
+        lReserved: LONG,
+        lResLength: LONG,
+        pbBuffer: LPBYTE,
+    ) -> HRESULT,
+}}
+STRUCT!{struct WIA_DATA_TRANSFER_INFO {
+    ulSize: ULONG,
+    ulSection: ULONG,
+    ulBufferSize: ULONG,
+    bDoubleBuffer: BOOL,
+    ulReserved1: ULONG,
+    ulReserved2: ULONG,
+    ulReserved3: ULONG,
+}}
+pub type PWIA_DATA_TRANSFER_INFO = *mut WIA_DATA_TRANSFER_INFO;
+STRUCT!{struct WIA_EXTENDED_TRANSFER_INFO {
+    ulSize: ULONG,
+    ulMinBufferSize: ULONG,
+    ulOptimalBufferSize: ULONG,
+    ulMaxBufferSize: ULONG,
+    ulNumBuffers: ULONG,
+}}
+pub type PWIA_EXTENDED_TRANSFER_INFO = *mut WIA_EXTENDED_TRANSFER_INFO;
+DEFINE_GUID!{IID_IWiaDataTransfer,
+    0xa6cef998, 0xa5b0, 0x11d2, 0xa0, 0x8f, 0x00, 0xc0, 0x4f, 0x72, 0xdc, 0x3c}
+RIDL!{#[uuid(0xa6cef998, 0xa5b0, 0x11d2, 0xa0, 0x8f, 0x00, 0xc0, 0x4f, 0x72, 0xdc, 0x3c)]
+interface IWiaDataTransfer(IIWiaDataTransferkVtbl): IUnknown(IUnknownVtbl) {
+    fn idtGetData(
+        pMedium: *const STGMEDIUM,
+        pIWiaDataCallback: *const IWiaDataCallback,
+    ) -> HRESULT,
+    fn idtGetBandedData(
+        pWiaDataTransInfo: PWIA_DATA_TRANSFER_INFO,
+        pIWiaDataCallback: *const IWiaDataCallback,
+    ) -> HRESULT,
+    fn idtQueryGetData(
+        pfe: *mut WIA_FORMAT_INFO,
+    ) -> HRESULT,
+    fn idtEnumWIA_FORMAT_INFO(
+        ppEnum: *mut *mut IEnumWIA_FORMAT_INFO,
+    ) -> HRESULT,
+    fn idtGetExtendedTransferInfo(
+        pExtendedTransferInfo: PWIA_EXTENDED_TRANSFER_INFO,
+    ) -> HRESULT,
+}}
+DEFINE_GUID!{IID_IWiaItem,
+    0x4db1ad10, 0x3391, 0x11d2, 0x9a, 0x33, 0x00, 0xc0, 0x4f, 0xa3, 0x61, 0x45}
+RIDL!{#[uuid(0x4db1ad10, 0x3391, 0x11d2, 0x9a, 0x33, 0x00, 0xc0, 0x4f, 0xa3, 0x61, 0x45)]
+interface IWiaItem(IWiaItemVtbl): IUnknown(IUnknownVtbl) {
+    fn GetItemType(
+        pItemType: LPLONG,
+    ) -> HRESULT,
+    fn AnalyzeItem(
+        lFlags: LONG,
+    ) -> HRESULT,
+    fn EnumChildItems(
+        ppIEnumWiaItem: *mut *mut IEnumWiaItem,
+    ) -> HRESULT,
+    fn DeleteItem(
+        lFlags: LONG,
+    ) -> HRESULT,
+    fn CreateChildItem(
+        lFlags: LONG,
+        bstrItemName: BSTR,
+        bstrFullItemName: BSTR,
+        ppIWiaItem: *mut *mut IWiaItem,
+    ) -> HRESULT,
+    fn EnumRegisterEventInfo(
+        lFlags: LONG,
+        pEventGUID: REFGUID,
+        ppIEnum: *mut *mut IEnumWIA_DEV_CAPS,
+    ) -> HRESULT,
+    fn FindItemByName(
+        lFlags: LONG,
+        bstrFullItemName: BSTR,
+        ppIWiaItem: *mut *mut IWiaItem,
+    ) -> HRESULT,
+    fn DeviceDlg(
+        hwndParent: HWND,
+        lFlags: LONG,
+        lIntent: LONG,
+        plItemCount: LPLONG,
+        ppIWiaItem: *mut *mut IWiaItem,
```

Comment:
```suggestion
        ppIWiaItem: *mut *mut *mut IWiaItem,
```
Wow, a wild triple pointer.